[SC-256] Uzupełnić wymagania

ajablonsk1 opened the pull request:
Teraz gdy prowadzący w aktualizacji wymagań poda np zły email to zwracam taki obiekt (ponizej x to email, ktory sie nie zgadza):

![image](https://user-images.githubusercontent.com/73038429/190260762-8767c688-7b2e-4cd6-a31e-e402c64f4606.png)
Działa to tak samo dla nazwy grupy, nazwy file taska i nazwy graphtaska.

Do tego do chapter/info dodałem info czy wymagania do tej aktywności zostały już dodane. Działa to tak, że jeżeli, zmieni np dateFrom na coś to od tego momentu te wymagania są brane jako nie defaultowe, nawet jeżeli potem by znowu ustawił dateFrom na nic. Sprawdza to tylko czy wymagania nie były ruszane przez prowadzącego, jeżeli coś zmienił to już jest uznane za "dodane". Przykład: 

localhost:8080/api/chapter/info?id=1

```
{
    "id": 1,
    "name": "Rozdział 1",
    "noActivities": 5,
    "noGraphTasks": 2,
    "noSurveyTasks": 1,
    "noInfoTasks": 1,
    "noFileTasks": 1,
    "maxPoints": 160.0,
    "mapSize": "8 x 5",
    "mapTasks": [
        {
            "id": 1,
            "posX": 5,
            "posY": 4,
            "type": "EXPEDITION",
            "title": "Dżungla kabli",
            "points": 60.0,
            "isFulfilled": false,
            "isCompleted": false,
            "areRequirementsAdded": true
        },
        {
            "id": 2,
            "posX": 2,
            "posY": 2,
            "type": "EXPEDITION",
            "title": "Dżungla kabli II",
            "points": 60.0,
            "isFulfilled": false,
            "isCompleted": false,
            "areRequirementsAdded": false
        },
        {
            "id": 3,
            "posX": 3,
            "posY": 3,
            "type": "TASK",
            "title": "Niszczator kabli",
            "points": 30.0,
            "isFulfilled": true,
            "isCompleted": false,
            "areRequirementsAdded": false
        },
        {
            "id": 4,
            "posX": 3,
            "posY": 0,
            "type": "INFO",
            "title": "Skrętki",
            "points": 0.0,
            "isFulfilled": true,
            "isCompleted": false,
            "areRequirementsAdded": false
        },
        {
            "id": 5,
            "posX": 7,
            "posY": 3,
            "type": "SURVEY",
            "title": "Example map feedback",
            "points": 10.0,
            "isFulfilled": true,
            "isCompleted": false,
            "areRequirementsAdded": false
        }
    ]
}
```
Tylko dla pierwszego taska dodałem wymagania. Dla drugiego dodałem, ale tylko defaultowe, dlatego jest false.

Jeszcze dodałem sprawdzenie czy nazwa grupy, email, nazwa graph taska oraz nazwa file taska nie zawierają ";". Jeżeli np przy rejestracji w mailu ktoś dał ";" to wywala reposne error z informacja:

![image](https://user-images.githubusercontent.com/73038429/190263363-b6704415-c91b-403e-81d2-272aeb18e178.png)

ajablonsk1 committed:
[SC-256] Added professor input for req validation and in chapter/info one field if the requirements are added

barto14753 commented on api/src/main/java/com/example/api/service/activity/task/TaskService.java:
```diff
@@ -258,35 +253,59 @@ public void addRequirementToActivity(ActivityRequirementForm form) throws Reques
-                            .stream()
-                            .filter(fileTask -> titles.contains(fileTask.getTitle()))
-                            .toList();
+                    List<FileTask> fileTasks = new LinkedList<>();
```
Jest jakiś powód dlaczego w tym casie zostało List<String> a w pozostałych zamieniłeś je na String[] ? Jeśli jest to bez róznicy zostawiłbym List<String>

ajablonsk1 replied:
intelij sie sral, ze moze bez sensu opakowywuje w liste xddd

barto14753 reviewed (approved):
Wygląda dobrze, ale obecnie mam jakiś problem z driverem postgresa i nie mogę odpalić nic odpalić. Daje approve żeby nie blokować.

dmncp commented:
> Wygląda dobrze, ale obecnie mam jakiś problem z driverem postgresa i nie mogę odpalić nic odpalić. Daje approve żeby nie blokować.

@barto14753 A nie możesz odpalić z dockera ? Bo możesz zpullować zmiany i wpisać `docker-compose up --build`

barto14753 commented:
Jak coś to odpaliłem i działa

ajablonsk1 committed:
[SC-256] Merge branch 'develop' into SC-256-uzupelnic-wymagania

# Conflicts:
#	api/src/main/java/com/example/api/error/exception/ExceptionMessage.java
